packages/graphql-toolbox/src/contexts/utils.ts

```diff
@@ -140,19 +145,41 @@ export const getDatabases = async (driver: neo4j.Driver): Promise<Neo4jDatabase[
     }
 };
 
+const getUrlSearchParam = (paramName: string): string | null => {
+    const queryString = window.location.search;
+    if (!queryString) return null;
+    const urlParams = new URLSearchParams(queryString);
+    return urlParams.get(paramName);
+};
+
+export const getConnectUrlSearchParamValue = (): {
+    url: string;
+    username: string;
+    protocol: string;
+} | null => {
+    const dbmsParam = getUrlSearchParam(CONNECT_URL_PARAM_NAME as string);
+    if (!dbmsParam) return null;
+    const [protocol, username, host] = dbmsParam.split(/:\/\/|@/);
```

Comment:
If the user is not passed in the URL, then the username variable will hold the host value and the host variable will hold undefined, breaking the feature.  
In my opinion the UserInfo part of the URL should be optional.

Reply:
Let me see if I can safeguard this better. To allow both urls with and without a username.
Good catch!